strings-csharp/HexToByteArray/Tests/Test.cs

```diff
@@ -4,85 +4,66 @@ namespace Tests
 {
     public class Test
     {
-        private readonly ConvertHex _convertHex;
         private const string HexString = "FF008000";
-        public Test()
-        {
-            _convertHex = new ConvertHex();
-        }
+        private readonly byte[] _expectedResult = { 255, 0, 128, 0 };
+        private readonly byte[] _expectedEmptyResult = new byte[0];
+
 
         [Fact]
         public void WhenValidHexPassed_ThenConverionSuccessful()
         {
-            byte[] expectedResult = { 255, 0, 128, 0 };
-            var result = _convertHex.FromHexString(HexString);
+            var result = ConvertHex.FromHexString(HexString);
 
-            Assert.Equal(expectedResult, result);
+            Assert.Equal(_expectedResult, result);
         }
 
         [Fact]
         public void WhenInvalidFormatHexPassed_ThenConverionFail()
         {
-            var expectedResult = "FormatException";
-            try
-            {
-                var result = _convertHex.FromHexString("Hello");
-            }
-            catch (Exception ex)
-            {
-                Assert.Equal(expectedResult, ex.GetType().Name);
-            }
+            var ex = Assert.Throws<FormatException>(() => ConvertHex.FromHexString("Hello"));
+
+            Assert.Equal("FormatException", ex.GetType().Name);
         }
 
         [Fact]
         public void WhenNullatHexPassed_ThenConverionFail()
         {
-            var expectedResult = "ArgumentNullException";
-            try
-            {
-                var result = _convertHex.FromHexString("");
-            }
-            catch (Exception ex)
-            {
-                Assert.Equal(expectedResult, ex.GetType().Name);
-            }
+            string? nullHexString = null;
```

Comment:
We can just pass null in to the method to check behavior in a null situation. Another note in relation to that is that with nullable enabled, the compiler is checking that stuff and using the annotations to surface that back to the caller of the method, so the method signature itself is conveying that null is not an acceptable value. When that is the case, we can skip checking the behavior around a null argument, as our API is saying "Null is not acceptable", which means results of passing in null are by definition "undefined".

Reply:
Makes sense. I have removed this test case. Thanks!